### PR TITLE
Add impl for converting rgb/rgba into hsl/hsla

### PR DESCRIPTION
**Pull request by vaidehijoshi**
Adds implementation for converting RGB and RGBA colors into HSL and HSLA.

### CODE REVIEW

**Comment by vaidehijoshi** [comment]
i rebased a commit into this PR that splits up Angles into it's own module, so am changing the base branch to make diffs more obvious

**Comment by vaidehijoshi** [comment]
@chancancode i think this is ok to merge, and then i can refactor how i handle the saturation & luminosity once i have impl'd percentages. let me know if you disagree, though.

File: src/lib.rs [hunk 1]
```diff
@@ -158,6 +285,19 @@ impl Color for RGBA {
...
         self
     }
+
+    fn to_hsl(self) -> HSL {
+        // convert to rgb, and then to hsl
+        let converted_rgb = self.to_rgb();
+        let HSL { h, s, l } = converted_rgb.to_hsl();
+
+        HSL::new(h, s, l)
+    }
```
Comment (chancancode) [inline src/lib.rs:291]: I think this whole method can just be `self.to_rgb().to_hsl()`

File: src/lib.rs [hunk 2]
```diff
@@ -347,9 +510,59 @@ mod css_color_tests {
 
         // FIXME: update these tests once HSL <-> RBG impl exists
```
Comment (chancancode) [inline src/lib.rs:507]: Pending

**Review by chancancode — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (vaidehijoshi) [reply src/lib.rs:507]: do you mean impl the hsl --> rgb in this pr itself?

Comment (chancancode) [reply src/lib.rs:507]: ah nvm I thought it was already working and you just forgot to remove this

Comment (vaidehijoshi) [reply src/lib.rs:507]: ok, cool :) going to merge this and then do hsl --> rgb next!

**Pull request merged**